### PR TITLE
Combine ingredients in shopping lists

### PR DESCRIPTION
**Pull request by siddharthist**
## Reifying units to a type
- This leads to a consistent internal representation. When dealing with units, we remove ambiguity at parse-time and never have to deal with synonyms internally.
- This has the disadvantage that units aren't presented exactly as the user entered them (arguably an advantage).
- It leads to more compile-time guarantees, when dealing with units GHC can now make sure our cases are exhaustive.

## Combining ingredients
This is the goal of the reification. If you have two units with the same names and units, say:
 - 1 (no unit) onion, large
 - 2 (no unit) onions, medium
It will be displayed in a shopping list as:
 - 3 onions, 1 large, 2 medium
So it is even smart about combining attributes in a way that the user can recover all the original data :smile: 

## Reorganization and cleanup
- `AddCLI` is definitely not a library module, it defines a UI
- Remove unnecessary modules from Cabal targets
- Remove doctests (you can just run `cabal doctest`, which will do everything this code currently does automatically)

### CODE REVIEW

**Comment by siddharthist** [comment]
Note that this introduces a backwards-incompatible change: units in already-defined recipes files will have to be modified.

File: app/herms.hs [hunk 1]
```diff
@@ -378,7 +377,7 @@ servingP t =  option auto
          <> short Str.servingShort
          <> help  (t Str.servingDesc)
          <> showDefault
-         <> value 0
+         <> value 1
```
Comment (siddharthist) [inline app/herms.hs]: Oh, I didn't meant to change this.

**Comment by siddharthist** [comment]
This isn't quite ready:
 - breaking changes should be noted in the documentation
 - the edit window needs to use `showUnit`, not `show`

**Comment by JackKiefer** [comment]
Wow! This is quite an update!!! I'm loving it! Is this ready to be merged?

**Comment by siddharthist** [comment]
Not quite yet! I will finish it soon, likely tomorrow. All those changes should be quite small (~15 lines total), so feel free to give it a review.

**Pull request merged**